website/source/docs/auth/gcp.html.md

```diff
@@ -10,51 +10,99 @@ description: |-
 
 # Google Cloud Auth Method
 
-The `gcp` auth method allows authentication against Vault using Google
-credentials. It treats Google Cloud Platform (GCP) as a Trusted Third Party and
-expects a [JSON Web Token][jwt] (JWT) signed by Google credentials from the
-authenticating entity. This token can be generated through different GCP APIs
-depending on the type of entity.
+The `gcp` auth method allows Google Cloud Platform (GCP) entities to
+authenticate to Vault. Vault treats GCP  as a Trusted Third Party and
+verifies authenticating entities against GCP APIs. This backend allows
+for authentication of:
+
+* IAM service accounts
+* Google Compute Engine (GCE) instances
+
+**NOTE**: This backend focuses on identities specific to Google Cloud and
+does not support authenticating Google users (i.e. `type: "authorized_user"`) 
+or OAuth against Google
 
 This plugin is developed in a separate GitHub repository at
 [`hashicorp/vault-plugin-auth-gcp`](https://github.com/hashicorp/vault-plugin-auth-gcp),
 but is automatically bundled in Vault releases. Please file all feature
 requests, bugs, and pull requests specific to the GCP plugin under that
 repository.
 
-## Authentication
+## Quick Start
+
+To authenticate, you will need two things:
+
+1. Proof-of-identity: 
+    * Service Account: A signed [JSON Web Token][jwt] (JWT), or GCP credentials 
+    with permissions to sign a JWT (see [`signJwt`][signjwt-method] IAM API 
+    method) for this service account
+    * Access to a metadata server for a GCE instance
+1. Name of a `role`, a binding of Vault policies and authZ 
+   restrictions preconfigured for the GCP auth backend that
+   an entity authenticates under. Roles are either type `iam`, which
+   supports either type of entity, or `gce`, which support only GCE instances.
 
-### Via the CLI
+### Via CLI
 
 The default path is `/gcp`. If this auth method was enabled at a different
 path, specify `-path=/my-path` in the CLI.
 
+The backend takes a signed [JSON Web Token][jwt] (JWT) as proof-of-identity.
+
 ```text
-$ vault login -method=gcp \
-    role="my-role" \
-    jwt="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
+$ vault login -method=gcp role="my-role" jwt="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
```

Comment:
Does this method accept a `jwt` parameter? I could not make this work when passing in my own JWT, every time it seemed like it tried to generate a new IAM JWT

Reply:
ah, you can use `vault write` but not `vault login` like so:

```
vault write -field=token auth/gcp/login role=my-role jwt=${JWT}
```

from https://cloud.google.com/solutions/using-vault-for-secret-management